newIDE/app/src/ObjectEditor/Editors/SpriteEditor/index.js

```diff
@@ -417,6 +481,196 @@ export default function SpriteEditor({
     [forceUpdate, onObjectUpdated, spriteConfiguration]
   );
 
+  const storageProvider = resourceManagementProps.getStorageProvider();
+  const resourceSources = resourceManagementProps.resourceSources
+    .filter(source => source.kind === 'image')
+    .filter(
+      ({ onlyForStorageProvider }) =>
+        !onlyForStorageProvider ||
+        onlyForStorageProvider === storageProvider.internalName
+    );
+
+  const importImages = React.useCallback(
+    async () => {
+      const resources = await resourceManagementProps.onChooseResource({
+        initialSourceName: resourceSources[0].name,
+        multiSelection: true,
+        resourceKind: 'image',
+      });
+      if (resources.length === 0) {
+        return;
+      }
+      resources.forEach(resource => {
+        applyResourceDefaults(project, resource);
+        project.getResourcesManager().addResource(resource);
+      });
+
+      addAnimations(groupResourcesByAnimations(resources));
+
+      // Important, we are responsible for deleting the resources that were given to us.
+      // Otherwise we have a memory leak, as calling addResource is making a copy of the resource.
+      resources.forEach(resource => resource.delete());
+
+      forceUpdate();
+
+      await resourceManagementProps.onFetchNewlyAddedResources();
+
+      if (onObjectUpdated) onObjectUpdated();
+    },
+    [
+      resourceManagementProps,
+      resourceSources,
+      addAnimations,
+      forceUpdate,
+      onObjectUpdated,
+      project,
+    ]
+  );
+
+  const adaptCollisionMaskIfNeeded = React.useCallback(
+    () => {
+      // If the first sprite of the first animation is updated,
```

Comment:
this comment fits better where the function is passed down below, because there's a check on animationIndex === 0

Reply:
+1